[Improvement] Add shellcheck and shfmt linting for shell scripts

## Objective
Add shellcheck and shfmt linting for all shell scripts in `scripts/` and the justfile shell recipes.

## Motivation
Issue #30 tracks adding Python linting and security scanning to CI, but there are two shell scripts (`scripts/scrape-agamemnon.sh`, `scripts/import-dashboards.sh`) plus inline bash in the justfile that have no linting. Shell scripts are prone to subtle bugs (unquoted variables, missing error handling) that shellcheck detects. shfmt ensures consistent formatting.

## Deliverables
- [ ] Add `shellcheck` to pixi.toml dependencies (or install in CI)
- [ ] Add `shfmt` to pixi.toml dependencies (or install in CI)
- [ ] Add a `just lint-shell` target that runs `shellcheck scripts/*.sh`
- [ ] Add shellcheck and shfmt steps to the CI workflow
- [ ] Fix any issues found by shellcheck in existing scripts

## Success Criteria
- [ ] `shellcheck scripts/*.sh` passes with zero warnings
- [ ] All shell scripts follow consistent formatting
- [ ] CI fails if new shell scripts introduce issues

## Priority
LOW -- improves script reliability, complements the Python linting in issue #30

---
*Updated per [ADR-006](https://github.com/HomericIntelligence/Odysseus/blob/main/docs/adr/006-decouple-from-ai-maestro.md) — ai-maestro references replaced with current architecture terminology.*

# Implementation Plan

Now I have all the context I need. Here's the implementation plan:

---

## Implementation Plan: Issue #46 — Shell Script Linting (shellcheck + shfmt)

### 1. Objective

Add shellcheck static analysis and shfmt formatting enforcement for `scripts/scrape-agamemnon.sh` and `scripts/import-dashboards.sh`. Wire both tools into a `just lint-shell` recipe and the existing CI `validate` job.

---

### 2. Approach

- **Dependency management**: Add `shellcheck` and `shfmt` under a new `[feature.dev.dependencies]` section in `pixi.toml` (keeping them dev-only, not runtime deps). Since pixi.toml currently has no feature sections at all, this requires adding the full feature block.
- **justfile recipe**: Add a single `lint-shell` recipe that runs both shellcheck and shfmt diff-check. Avoid heredocs per the `just-recipe-heredoc-fix` team knowledge (use multi-command recipe lines or `&&` chains instead).
- **CI**: Add a new `lint-shell` job to `.github/workflows/ci.yml` that installs shellcheck/shfmt directly (via `apt-get` or GitHub Actions marketplace — no pixi needed in CI since pixi isn't used in the existing workflow). Apply `ci-workflow-hardening-checklist` patterns: `permissions: contents: read`, `timeout-minutes: 10`, `concurrency` group.
- **Fix violations first**: Run shellcheck locally against both scripts before enabling CI gate, fix any issues in the same PR. Both scripts already use `set -euo pipefail` and look clean, but shellcheck may flag edge cases (e.g., the `echo -e` usage in scrape-agamemnon.sh is not portable — shellcheck SC2028/SC2039).

---

### 3. Files to Create

None — no new files are required.

---

### 4. Files to Modify

#### `pixi.toml`
Add a `[feature.dev.dependencies]` section with shellcheck and shfmt:

```toml
[feature.dev.dependencies]
shellcheck = ">=0.9"
shfmt      = ">=3.7"

[environments]
dev = { features = ["dev"], solve-group = "default" }
```

#### `justfile`
Add a `lint-shell` recipe after the existing `import-dashboards` recipe:

```just
# Lint shell scripts with shellcheck and check formatting with shfmt
lint-shell:
    shellcheck scripts/*.sh
    shfmt -d scripts/*.sh
```

No heredocs, no multi-line constructs — safe per `just-recipe-heredoc-fix` guidance.

Also add a `format-shell` recipe for fixing formatting in-place:

```just
# Format shell scripts in-place with shfmt
format-shell:
    shfmt -w scripts/*.sh
```

#### `.github/workflows/ci.yml`
Add a new `lint-shell` job (parallel to `validate`, not a step inside it):

```yaml
lint-shell:
  name: Lint shell scripts
  runs-on: ubuntu-latest
  permissions:
    contents: read
  timeout-minutes: 10
  concurrency:
    group: lint-shell-${{ github.workflow }}-${{ github.head_ref || github.sha }}
    cancel-in-progress: true
  steps:
    - uses: actions/checkout@v4
    - name: Install shellcheck and shfmt
      run: |
        sudo apt-get install -y shellcheck
        sudo curl -sSL https://github.com/mvdan/sh/releases/download/v3.7.0/shfmt_v3.7.0_linux_amd64 \
          -o /usr/local/bin/shfmt && sudo chmod +x /usr/local/bin/shfmt
    - name: shellcheck
      run: shellcheck scripts/*.sh
    - name: shfmt (diff check)
      run: shfmt -d scripts/*.sh
```

> **Note on shfmt install**: `shfmt` is not in Ubuntu's default apt repositories. The recommended approach is downloading the binary directly from the GitHub releases page (mvdan/sh). The URL above uses a pinned version `v3.7.0` — pin to a specific release for reproducibility.

#### `scripts/scrape-agamemnon.sh` (if violations found)
Expected potential shellcheck findings:
- **SC2028**: `echo -e` color escape sequences — shellcheck may flag `echo -e` as non-portable. Fix: replace with `printf` for color output, or add `# shellcheck disable=SC2028` with justification if `echo -e` is intentional.
- No other obvious issues given the script's clean structure.

#### `scripts/import-dashboards.sh` (if violations found)
Script looks clean. Shellcheck may flag nothing — both scripts already follow best practices.

---

### 5. Implementation Order

1. **Run shellcheck locally against both scripts** to discover actual violations before touching CI:
   ```bash
   shellcheck scripts/scrape-agamemnon.sh scripts/import-dashboards.sh
   ```
2. **Fix any shellcheck violations** in the scripts (likely `echo -e` → `printf` in `scrape-agamemnon.sh`).
3. **Run shfmt diff check** to see if formatting changes are needed:
   ```bash
   shfmt -d scripts/*.sh
   ```
4. **Apply shfmt formatting** if needed:
   ```bash
   shfmt -w scripts/*.sh
   ```
5. **Edit `pixi.toml`** — add `[feature.dev.dependencies]` and `[environments]` sections.
6. **Edit `justfile`** — add `lint-shell` and `format-shell` recipes.
7. **Edit `.github/workflows/ci.yml`** — add `lint-shell` job.
8. **Verify end-to-end**: `just lint-shell` should exit 0 with no output.

---

### 6. Verification

| Check | Command | Expected Result |
|-------|---------|-----------------|
| shellcheck passes locally | `shellcheck scripts/*.sh` | Zero warnings, exit 0 |
| shfmt reports no diffs | `shfmt -d scripts/*.sh` | No output, exit 0 |
| just recipe works | `just lint-shell` | Both tools run, exit 0 |
| CI job structure valid | `yamllint .github/workflows/ci.yml` | No errors |
| No regressions | Existing `validate` job still passes | All YAML/JSON checks green |

---

### 7. Skills Used

**Invoked during planning:**
- `hephaestus:advise` — searched team knowledge base for prior learnings before planning

**Team knowledge base skills referenced (from Prior Learnings):**
- `ci-cd-pipeline-maintenance-patterns` — parallel lint job structure, pixi + concurrency controls pattern
- `enable-yaml-markdown-linting` — exact parallel pattern (yamllint as pixi dep + just recipe + CI step); this plan mirrors that approach for shellcheck/shfmt
- `just-recipe-heredoc-fix` — **critical**: avoided heredocs in `lint-shell` recipe body; used simple single-line commands instead
- `ci-workflow-hardening-checklist` — applied `permissions: contents: read`, `timeout-minutes: 10`, `concurrency` group with `|| github.sha` fallback

---
*Generated by Claude Code Planner*

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -72,3 +72,6 @@ jobs:
               sys.exit(1)
           print("SUCCESS: No HIGH severity security issues found")
           SCAN_CHECK
+
+      - name: Lint shell scripts
+        run: shellcheck --severity=warning scripts/*.sh

--- a/scripts/backup.sh
+++ b/scripts/backup.sh
@@ -22,4 +22,4 @@ done
 echo ""
 echo "Backup complete: $BACKUP_DIR"
 echo "Files:"
-ls -lh "$BACKUP_DIR/${TIMESTAMP:0:8}"* 2>/dev/null || echo "  (no backups found with today's date)"
+ls -lh "${BACKUP_DIR}/${TIMESTAMP:0:8}"* 2>/dev/null || echo "  (no backups found with today's date)"

